### PR TITLE
Made FlowCheckpointVersionNodeStartupCheckTest leaner

### PR DESCRIPTION
**Pull request by shamsasari**
* Removed `restart node successfully with suspended flow` as it duplicates `TraderDemoTest#Test restart node during flow works properly`
* Removed the need for a notary

### CODE REVIEW

File: samples/trader-demo/workflows-trader/src/integration-test/kotlin/net/corda/traderdemo/TraderDemoTest.kt [hunk 1]
```diff
@@ -85,27 +91,20 @@ class TraderDemoTest {
...
-            nodeARpc.stateMachinesFeed().updates.toBlocking().first() // wait until initiated flow starts
-            nodeA.stop()
-            startNode(providedName = DUMMY_BANK_A_NAME, rpcUsers = listOf(demoUser), customOverrides = mapOf("p2pAddress" to nodeA.p2pAddress.toString()))
-            stxFuture.getOrThrow()
+            val (buyer, seller, bank) = listOf(
+                    startNode(providedName = DUMMY_BANK_A_NAME),
+                    startNode(providedName = DUMMY_BANK_B_NAME),
+                    startNode(providedName = BOC_NAME)
+            ).map { it.getOrThrow() }
+            TraderDemoClientApi(bank.rpc).runIssuer(amount = 100.DOLLARS, buyerName = DUMMY_BANK_A_NAME, sellerName = DUMMY_BANK_B_NAME)
```
Comment (tudor-malene) [inline samples/trader-demo/workflows-trader/src/integration-test/kotlin/net/corda/traderdemo/TraderDemoTest.kt]: Easier to follow the test now.

File: .idea/compiler.xml [hunk 2]
```diff
@@ -4,6 +4,11 @@
     <bytecodeTargetLevel target="1.8">
       <module name="api-scanner_main" target="1.8" />
       <module name="api-scanner_test" target="1.8" />
+      <module name="attachment-demo-contracts_main" target="1.8" />
```
[the remark this answers is not in the document]
Comment (shamsasari) [reply .idea/compiler.xml]: Generated by IntelliJ, due to the Kasia’s changes I think.

File: node/src/integration-test/kotlin/net/corda/node/flows/FlowCheckpointVersionNodeStartupCheckTest.kt [hunk 3]
```diff
@@ -1,133 +1,87 @@
...
-                if (r.consumed.isNotEmpty()) r.consumed.first() else r.produced.first()
-            }
-            assertNotNull(result)
-            assertEquals(message, result.state.data.message)
-        }
+        val defaultCordapp = enclosedCordapp()
     }
 
     @Test
     fun `restart node with incompatible version of suspended flow due to different jar name`() {
```
[the remark this answers is not in the document]
Comment (shamsasari) [reply node/src/integration-test/kotlin/net/corda/node/flows/FlowCheckpointVersionNodeStartupCheckTest.kt]: It’s actually a feature. I was thinking wheather I should relax it but that’s for a different Pr

[the remark this answers is not in the document]
Comment (tudor-malene) [reply node/src/integration-test/kotlin/net/corda/node/flows/FlowCheckpointVersionNodeStartupCheckTest.kt]: Cool. I guess we should get to the bottom of this in a different PR. I don't think the filename should matter.

**Pull request merged**